src/lib/y2network/dialogs/wireless_wep_keys.rb

```diff
@@ -0,0 +1,253 @@
+            next unless key
+            Item(Id(i), i.to_s, key, i == @settings.default_key ? "X" : "")
+          end
+
+          Yast::UI.ChangeWidget(Id(:wep_keys_table), :Items, table_items.compact)
+          [:wep_keys_delete, :wep_keys_edit, :wep_keys_default].each do |k|
+            Yast::UI.ChangeWidget(
+              Id(k),
+              :Enabled,
+              !@settings.keys.compact.empty?
+            )
+          end
+          Yast::UI.ChangeWidget(
+            Id(:wep_keys_add),
+            :Enabled,
+            @settings.keys.compact.size < 4 # only 4 keys are possible
+          )
+        end
+
+        # @return [Symbol, nil] dialog result
+        def handle(event)
+          return nil if event["EventReason"] != "Activated"
+
+          cur = Yast::UI.QueryWidget(Id(:wep_keys_table), :CurrentItem).to_i
+          case event["ID"]
+          when :wep_keys_edit
+            key = dialog(@settings.keys[cur])
+            if key
+              @settings.keys[cur] = key
+              refresh_table
+              Yast::UI.ChangeWidget(Id(:wep_keys_table), :CurrentItem, cur)
+            end
+          when :wep_keys_add
+            key = dialog
+            if key
+              # replace first nil value
+              index = @settings.keys.index(nil)
+              if index
+                @settings.keys[index] = key
+              else
+                @settings.keys << key
+              end
+              log.info "new keys #{@settings.keys.inspect}"
+              refresh_table
+              Yast::UI.ChangeWidget(
+                Id(:wep_keys_table),
+                :CurrentItem,
+                @settings.keys.compact.size - 1
+              )
+            end
+          when :wep_keys_delete
+            @settings.keys.delete_at(cur)
+            refresh_table
+          when :wep_keys_default
+            @settings.default_key = cur
+            refresh_table
+            Yast::UI.ChangeWidget(Id(:wep_keys_table), :CurrentItem, cur)
+          end
+
+          nil
+        end
+
+        # Open a dialog to add/edit a key.
+        # TODO: own class for it
+        # @param value  [String, nil] existing key to edit or nil for new key.
+        # @return      [String, nil] key or nil if dialog is canceled
+        def dialog(value = nil)
+          value ||= ""
+          Yast::UI.OpenDialog(
+            Opt(:decorated),
+            VBox(
+              HSpacing(1),
+              # TextEntry label
+              TextEntry(Id(:key), _("&WEP Key"), value),
+              HSpacing(1),
+              HBox(
+                PushButton(Id(:ok), Opt(:default), Yast::Label.OKButton),
+                PushButton(Id(:cancel), Yast::Label.CancelButton)
+              )
+            )
+          )
+
+          Yast::UI.SetFocus(Id(:key))
+          val = nil
+          while (ret = Yast::UI.UserInput) == :ok
+            val = Yast::UI.QueryWidget(Id(:key), :Value)
+            if !valid_key(val)
```

Comment:
What about:

```ruby
break if valid_key(val)
Yast::Popup.Error("...")
```

Reply:
it is possible. Originally there was more validation methods, but I keep only this one.